src/gt4py/next/type_system/type_info.py

```diff
@@ -642,7 +641,10 @@ def return_type_field(
             new_dims.append(d)
         else:
             new_dims.extend(target_dims)
-    return ts.FieldType(dims=new_dims, dtype=field_type.dtype)
+    return ts.FieldType(
+        dims=sorted(new_dims, key=lambda dim: (common.dims_kind_order[dim.kind], dim.value)),
```

Comment:
```suggestion
        dims=_ordered_dims(new_dims),
```
I'm surprised this is needed, in what case is the order not preserved?

Reply:
This is necessary because of the following in test_gt4py_builtins.py:
```
@gtx.field_operator
def reduction_ek_field(
    edge_f: common.Field[[Edge, KDim], np.int32],
) -> common.Field[[Vertex, KDim], np.int32]:
    return neighbor_sum(edge_f(V2E), axis=V2EDim)
```
Otherwise, one gets a: ` ValueError: Dimensions 'Vertex[horizontal], V2E[local], KDim[vertical]' are not ordered correctly, expected 'Vertex[horizontal], KDim[vertical], V2E[local]'.`